alicloud/resource_alicloud_instance.go

```diff
@@ -651,13 +653,18 @@ func resourceAliyunInstanceUpdate(d *schema.ResourceData, meta interface{}) erro
 	if err != nil {
 		return WrapError(err)
 	}
-	if imageUpdate || vpcUpdate || passwordUpdate || typeUpdate {
+	target := ""
+	statusUpdate := d.HasChange("status")
+	if statusUpdate {
+		target = d.Get("status").(string)
+	}
+	if imageUpdate || vpcUpdate || passwordUpdate || typeUpdate || statusUpdate {
 		run = true
 		instance, errDesc := ecsService.DescribeInstance(d.Id())
 		if errDesc != nil {
 			return WrapError(errDesc)
 		}
-		if instance.Status == string(Running) {
+		if instance.Status == string(Running) || (instance.Status != target && target == string(Stopped)) {
```

Comment:
I think the line 667 should be `if (statusUpdate && target == string(Stopped)) || instance.Status == string(Running) {`